src/rdpc/analysesProcessor.ts

```diff
@@ -69,6 +115,50 @@ export const analysisStream = async function* ({
   }
 };
 
+export const aggregateSpecimensByDonorId = (
+  analyses: AnalysisWithSpecimens[]
+): SpecimensByDonors => {
+  const result = analyses.reduce<SpecimensByDonors>((acc, analysis) => {
+    const donorWithSpecimens = analysis.donors.reduce<SpecimensByDonors>(
+      (_acc, donor) => {
+        _acc[donor.donorId] = donor.specimens;
+        return _acc;
+      },
+      {}
+    );
+
+    Object.entries(donorWithSpecimens).forEach(([donorId, specimens]) => {
+      const existing = acc[donorId];
+      const mergedSpcimens =
+        existing === undefined
+          ? specimens
+          : mergedSpecimens(specimens, existing);
+      acc[donorId] = mergedSpcimens;
+    });
+    return acc;
+  }, {});
+
+  return result;
+};
+
+/**
+ * Removes duplicate specimens in merged arrays
+ * @param existingSpecimens
+ * @param toMerge
+ */
+export const mergedSpecimens = (
+  existingSpecimens: Specimen[],
+  toMerge: Specimen[]
```

Comment:
Is there any difference between `existingSpecimens` and `toMerge` ? If i switch the variable order in the method do I get a different result?

Reply:
the function will always return the same merged result regardless of variable orders, I tested locally as well